controllers/subscription.ts

```diff
@@ -0,0 +1,89 @@
+import { CustomRequest, CustomResponse } from "../types/global";
+const { addOrUpdate } = require("../models/users");
+const { INTERNAL_SERVER_ERROR } = require("../constants/errorMessages");
+const nodemailer = require("nodemailer");
+const config = require("config");
+const emailSubscriptionCredentials = config.get("emailSubscriptionCredentials");
+
+export const subscribe = async (req: CustomRequest, res: CustomResponse) => {
+  const dev = req.query.dev === "true";
+    if (!dev) {
+      return res.boom.notFound("Route not found");
+    }
+    const { email } = req.body;
+    const phoneNumber = req.body.phoneNumber || null;
+    const userId = req.userData.id;
+    const data = { email, isSubscribed: true, phoneNumber };
+    const userAlreadySubscribed = req.userData.isSubscribed;
+  try {
+    if (userAlreadySubscribed) {
+      return res.boom.badRequest({message: "User is already subscribed"});
+    }
+    await addOrUpdate(data, userId);
+    return res.status(201).json({
+      message: "user subscribed successfully",
+    });
+  } catch (error) {
+    res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+export const unsubscribe = async (req: CustomRequest, res: CustomResponse) => {
```

Comment:
1. The dev check at the beginning of the function might be unnecessary or misplaced. If this is truly a development-only route, consider moving this check to the route definition rather than within the handler function.
2. Inconsistent error response format: In the if (!userAlreadySubscribed) check, the error response was using an object with a message property, while other error responses were using strings directly.
3. Inconsistent return in the catch block: The original code was using res.boom.badImplementation() without a return statement. This could lead to the function continuing execution after sending an error response.
4. Potential issue with error handling: The current error handling doesn't provide any specific information about the error. Depending on your requirements, you might want to log the error or provide more detailed error information (while being careful not to expose sensitive data).
5. Minor typo: "user unsubscribed successfully" has been capitalized to "User unsubscribed successfully" for consistency with other messages.

Reply:
1. Added a middleware at route level for handling dev flag.
2. Now, we are only passing string in the message, as we only need the response.ok on the frontend. But for  **sendEmail** function it is returning message as well.
3. Added return statement to catch block as well.
4. Added logger while handling error.
5. Fixed this.